src/Compiler/Checking/TailCallChecks.fs

```diff
@@ -332,7 +316,7 @@ and CheckExpr (cenv: cenv) (env: env) origExpr (ctxt: PermitByRefExpr) (isTailCa
 
         | Expr.Const (_, _m, _ty) -> ()
 
-        | Expr.Val (vref, vFlags, m) -> CheckValUse cenv env (vref, vFlags, m) ctxt isTailCall
+        | Expr.Val (_vref, _vFlags, _m) -> ()
```

Comment:
Nit: ignore the whole thing: `| Expr.Const _ | Expr.Val _ -> ()`

Reply:
Thanks, yeah, most likely there's still a lot of cleanup/code removal possible I think.